src/main/java/com/awesomeorg/airlineservice/service/TicketService.java

```diff
@@ -1,12 +1,103 @@
 package com.awesomeorg.airlineservice.service;
 
+import com.awesomeorg.airlineservice.entity.Ticket;
+import com.awesomeorg.airlineservice.exceptions.TicketAvailabilityExceededException;
+import com.awesomeorg.airlineservice.exceptions.TicketNotFoundException;
+import com.awesomeorg.airlineservice.protocol.CreateTicketRequest;
+import com.awesomeorg.airlineservice.protocol.UpdateTicketRequest;
 import com.awesomeorg.airlineservice.repository.TicketRepository;
 import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
 import org.springframework.stereotype.Service;
 
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
 @Service
 @RequiredArgsConstructor
 public class TicketService {
 
     private final TicketRepository ticketRepository;
-}
+
+    public Optional<Ticket> findTicketById(final Long ticketId) {
+        if (ticketId == null) {
+            return Optional.empty();
+        }
+        return ticketRepository.findById(ticketId);
+    }
+
+    public Page<Ticket> findFreeTicket(CreateTicketRequest query, PageRequest pageRequest) {
+        LocalDate dateOfFlight = query.getDateOfFlight();
+        return ticketRepository.findFreeTicket(dateOfFlight,pageRequest);
+    }
+
+
+
+    public List<Ticket> createTickets(CreateTicketRequest request) {
+        LocalDate dateOfPurchase = LocalDate.now();
+
+        final Page<Ticket> existingTickets = ticketRepository.findFreeTicket(dateOfPurchase, PageRequest.of(0, Integer.MAX_VALUE));
```

Comment:
For the future: it's better to create a new query that would just count and returns the value straight away, since you are creating a lot of unnecessary objects. Although, here it would be okay and you don't need to do anything